lib/Doctrine/Migrations/Tools/Console/Command/DoctrineCommand.php

```diff
@@ -75,6 +92,26 @@ protected function initialize(InputInterface $input, OutputInterface $output) :
             $this->dependencyFactory->setConfigurationLoader($configurationLoader);
         }
 
+        if ($input->getOption('em') !== null && $input->getOption('conn') !== null) {
+            throw new InvalidOptionUsage('You can specify only one of the --em and --conn options.');
+        }
+
+        $connectionLoader = $this->dependencyFactory->getConnectionLoader();
+        if ($connectionLoader instanceof ConnectionRegistryConnection && $input->getOption('conn') !== null) {
```

Comment:
I'm not really happy about this conditionals based on concrete classes... any ideas on how to avoid this?

@greg0ire @SenseException ?

Reply:
It's like `ConnectionRegistryConnection` isn't supposed to be a `ConnectionLoader`. The `DependencyFactory` is already too crowded for implementing new methods there.